draft-ietf-tls-esni.md

```diff
@@ -804,6 +804,19 @@ by attackers on the local network, which is a common case where ClientHello and
 encryption are desired. Moreover, as noted in the introduction, SNI encryption is
 less useful without encryption of DNS queries in transit via DoH or DPRIVE mechanisms.
 
+## Client Tracking
+
+A malicious client-facing server could distribute unique, per-client ECHOConfig
+structures as a way of tracking clients across subsequent connections. On-path
+adversaries which know about these unique keys could also track clients in this
+way by observing TLS connection attempts.
+
+The cost of this type of attack scales linearly with the desired number of target
+clients. Moreover, DNS caching behavior makes targeting individual users for extended
+periods of time, e.g., using per-client ECHOConfig structures delivered via HTTPSSVC
+RRs with high TTLs, challenging. Also, clients can help mitigate this problem by
+flushing any ECHOConfig state upon changing networks.
```

Comment:
Is the thinking here that the client might have some ECHOConfig state outside the DNS cache, or is this just saying the client could flush its DNS cache on network change? An attacker that can distribute per-client ECHOConfigs in DNS could just as easily distribute per-client A/AAAA records in DNS. (Well, A may be hard due to address scarcity, but AAAA should be doable.) So if you're concerned about this attack, you probably want to treat this as a DNS-wide issue and drop the entire DNS cache.

Reply:
Indeed -- in practice it'll likely be the entire cache. I didn't want to include other state (V4 or V6) to make any assumption on what the cache might contain, be it the DNS cache or something else. Is this just too generic for your taste?

Reply:
Mostly a clarifying question. Though perhaps we should say "flushing any DNS state" instead of flushing any ECHOConfig state". Or perhaps "flushing any DNS or ECHOConfig state". Or maybe we leave it along because too many words. :-)

Reply:
How about: "flushing any ECHOConfig state, which may the include DNS cache"?

Reply:
SGTM